falcon/api.py

```diff
@@ -556,7 +558,15 @@ def handle(req, resp, ex, params):
 
         # Insert at the head of the list in case we get duplicate
         # adds (will cause the most recently added one to win).
-        self._error_handlers.insert(0, (exception, handler))
+        try:
+            exception_tuple = tuple(exception)
+        except TypeError:
+            exception_tuple = tuple([exception, ])
+
+        if all(issubclass(exc, BaseException) for exc in exception_tuple):
+            self._error_handlers.insert(0, (exception_tuple, handler))
```

Comment:
I realized there is a minor performance issue lurking here: for single exception classes (not tuples), we may want to insert just `exception`, not a tuple with one element.

The penalty is very minor (~10 ns), OTOH we would hit it every time when handling an error.

Reply:
solved in 73b7d64